### PR TITLE
DownloadPackageV1 - Fix successful result status on download failure

### PR DESCRIPTION
**Pull request by ByAgenT**
**Task name**: DownloadPackageV1

**Description**: Fix a bug when failure to download a package was marked successful

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) Y (Test to cover case when download fails)

**Attached related issue:** (Y/N) [Link](https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1818047)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

Tested both valid and invalid version case on a local pipeline:
image
image

### CODE REVIEW

File: Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+tr.setInput("extract", "true");
+tr.setInput("verbosity", "verbose");
+
+// Set variables.
+process.env["AGENT_TEMPDIRECTORY"] = tempPath;
+process.env["SYSTEM_TEAMFOUNDATIONCOLLECTIONURI"] = "https://abc.visualstudio.com/";
+process.env["AGENT_VERSION"] = "2.116.0";
+process.env["HOME"] = "/users/test";
+process.env["ENDPOINT_AUTH_SYSTEMVSSCONNECTION"] =
+    '{"scheme":"OAuth","parameters":{"AccessToken":"YWFtYWxsYWQ6ZXd0emE1bmN3MzN6c3lyM2NoN2prazUzejczamN6MnluNGtiNzd0ZXc0NnlhZzV2d3ZlcQ=="}}';
```
Comment (satbai) [inline Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts:27]: Where did you get this access token? We shouldn't check in any access tokens

File: Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+tr.setInput("feed", "feedId");
+tr.setInput("view", "viewId");
+tr.setInput("definition", "6f598cbe-a5e2-4f75-aa78-e0fd08301a15");
+tr.setInput("version", "versionId");
+tr.setInput("downloadPath", outputPath);
+tr.setInput("extract", "true");
+tr.setInput("verbosity", "verbose");
+
+// Set variables.
+process.env["AGENT_TEMPDIRECTORY"] = tempPath;
```
Comment (satbai) [inline Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts:22]: I'm not super familiar with the way the pipeline task l0 tests are set up to be honest - are these values something all of the l0s have set?

Comment (ByAgenT) [reply Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts:22]: I copied this setup from another test case in DownloadPackageV1. To the access token mentioned above, it is already checked in sibling tests, but it is not necessary for the test to finish, so I'll remove that here. But if I remove other env variables, tests are starting to fail.

File: Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts (hunk 1 above)
Comment (satbai) [reply Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts:27]: @aasim do you have context as I think you're the most familiar with the DownloadPackage task? As @ByAgenT  mentioned above it looks like this is just the way we've set up the l0s.

Comment (ByAgenT) [reply Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts]: I just cleared variables that are not affecting test results for all tests in DownloadPackageV1, by the way.

**Review by satbai — APPROVED** [review]

File: Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts (hunk 2 above)
Comment (aasim) [reply Tasks/DownloadPackageV1/Tests/L0DownloadNpmPackageDownloadError.ts:22]: Yeah, you can remove them as long as tests aren't failing... some of these variables are used in the test.

**Review by aasim — APPROVED** [review]

**Pull request merged**